### PR TITLE
DC-3204: added auditing on error handler

### PR DESCRIPTION
**Pull request by matthewgroom**

### CODE REVIEW

File: app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala [hunk 1]
```diff
@@ -60,19 +60,22 @@ class SecureMessageController @Inject()(
               .createConversation(conversation)
               .map {
                 case Right(_) =>
-                  val _ = auditCreateConversation(EventTypes.Succeeded, conversation)
+                  val _ = auditCreateConversation(EventTypes.Succeeded, conversation, "Conversation Created")
                   Created
                 case Left(error: SecureMessageError) =>
-                  val _ = auditCreateConversation(EventTypes.Failed, conversation)
+                  val _ = auditCreateConversation(EventTypes.Failed, conversation, error.message)
```
Comment (OlegEfrem) [inline app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala:66]: You don't need to call auditing here as our error handler calls bootstrap error handler which logs and audits the events: https://github.com/hmrc/secure-message/blob/master/app/uk/gov/hmrc/securemessage/ErrorHandler.scala#L32

Comment (matthewgroom) [reply app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala:66]: okay, I'll take all of this out

Comment (matthewgroom) [reply app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala:66]: Will this be logged in Splunk explicitly then?

File: conf/application.conf [hunk 2]
```diff
@@ -110,7 +110,7 @@ metrics {
 # Microservice specific config
 
 auditing {
-  enabled = true
+  enabled = false
```
Comment (elabeca) [inline conf/application.conf:113]: secret sauce lol

**Review by elabeca — APPROVED** [review]

File: app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala (hunk 1 above)
Comment (OlegEfrem) [reply app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala]: Yes, look at the implementation of the error logger from hmrc library we call in our error handler, they call both audit and log

File: app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala [hunk 3]
```diff
@@ -60,19 +60,19 @@ class SecureMessageController @Inject()(
               .createConversation(conversation)
               .map {
                 case Right(_) =>
-                  val _ = auditCreateConversation(EventTypes.Succeeded, conversation)
+                  val _ = auditCreateConversation(EventTypes.Succeeded, conversation, "Conversation Created")
```
Comment (muralishub) [inline app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala:63]: we dont need val _ ?

File: app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala [hunk 4]
```diff
@@ -60,19 +60,19 @@ class SecureMessageController @Inject()(
...
                   handleErrors(ClientName.withName(conversation.client), conversation.id, error)
               }
               .recover {
-                case NonFatal(error) => handleErrors(ClientName.withName(conversation.client), conversation.id, error)
+                case NonFatal(error) =>
+                  handleErrors(ClientName.withName(conversation.client), conversation.id, error)
               }
           }
-        case _ => Future(handleErrors(client, conversationId, InvalidRequest(s"Not supported client: $client")))
+        case _ =>
```
Comment (muralishub) [inline app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala:73]: same here

File: conf/application.conf (hunk 2 above)
Comment (matthewgroom) [reply conf/application.conf:113]: lol

**Review by muralishub — APPROVED** [review]

**Pull request closed without merging**